modules/light-clients/08-wasm/types/wasm_test.go

```diff
@@ -0,0 +1,92 @@
+package types_test
+
+import "github.com/cosmos/ibc-go/modules/light-clients/08-wasm/types"
+
+func (suite *TypesTestSuite) TestGetCodeHashes() {
+	testCases := []struct {
+		name      string
+		malleate  func()
+		expResult func(codeHashes types.CodeHashes)
+	}{
+		{
+			"success: empty code hashes",
+			func() {},
+			func(codeHashes types.CodeHashes) {
+				suite.Require().Empty(codeHashes.CodeHashes)
+			},
+		},
+		{
+			"success: non-empty code hashes",
+			func() {
+				types.AddCodeHash(suite.chainA.GetContext(), GetSimApp(suite.chainA).AppCodec(), []byte("codehash"))
+			},
+			func(codeHashes types.CodeHashes) {
+				suite.Require().Len(codeHashes.CodeHashes, 1)
+				suite.Require().Equal([]byte("codehash"), codeHashes.CodeHashes[0])
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+			tc.malleate()
+
+			codeHashes := types.GetCodeHashes(suite.chainA.GetContext(), GetSimApp(suite.chainA).AppCodec())
+			tc.expResult(codeHashes)
+		})
+	}
+}
+
+func (suite *TypesTestSuite) TestAddCodeHash() {
```

Comment:
just a thought, currently `AddCodeHash` does not protect against adding a duplicate code hash. We test for this before we use the add function, but maybe it's worth considering panicking in this fn if a duplicate is going to be added?

I don't feel super strongly about this, so if we want to keep the store access fns as dumb as possible I'm happy with that too.

Reply:
I think I remember it being idempotent so a dup hash wouldn't be an issue (someone will just pay some more for it). Will look into it again tho!

Reply:
It increases gas cost for every initialization after that as well (anytime `HasCodeHashes` or `GetCodeHashes` is called), but it's not so much more gas for it to be an issue (since gov needs approve anyways)